internal/producer/src/accounts/accounts.go

```diff
@@ -277,6 +277,43 @@ func MintAurumUpdateAccountBalanceTable(dbConnection *sql.DB, pkhash []byte, val
 	return errors.New("Failed to find row")
 }
 
+func ValidateContract(c *Contract, table string, authorizedMinters [][]byte) (bool, error) {
+	db, err := sql.Open("sqlite3", table)
+	if err != nil {
+		return false, errors.New("Failed to open table")
+	}
+	defer db.Close()
+
+	// check for zero value transaction
+	if c.Value == 0 {
+		return false, nil
+	}
+
+	// check for unauthorized minting contracts
+	if c.SenderPubKey == nil {
+		validMinting := false
+		for _, mintersPubKHash := range authorizedMinters {
+			if bytes.Equal(c.RecipPubKeyHash, mintersPubKHash) {
+				validMinting = true
+				break
+			}
+		}
+		if !validMinting {
+			// unauthorized minting
+			return false, nil
+		}
+
+		// authorized minting
+		err = MintAurumUpdateAccountBalanceTable(db, c.RecipPubKeyHash, c.Value)
+		if err != nil {
+			return false, err
```

Comment:
Can you return a custom error here

Reply:
Oh, the reason that I did not return a custom error here is because `MintAurumUpdateAccountBalanceTable` already returns custom errors so I felt that it might be a bit more helpful if I return the func's custom errors instead of making a new custom error.

Reply:
That's a good point, can you nest the error in the returned error? we can build a custom stack trace that way. the way to do it would be `errors.New("some custom error message: "+err.Error())`, with `err` being the `err` returned from the update function.

Reply:
Sure thing